### PR TITLE
create custom exceptions

### PR DESCRIPTION
**Pull request by cli1903**

### CODE REVIEW

**Review by KoopaKing — COMMENTED** [review]
You should have all of these extend a custom exception type. This should allow them to be caught together by the error translator. Something like

```java
public abstract class TripMeOutException extends Exception {
  public abstract int restStatusCode();
}

public class PlaceVisitNotFoundException extends Exception {
  public PlaceVisitNotFoundException(String errorMsg) {
    super(errorMsg);
  }

  public PlaceVisitNotFoundException(String errorMsg, Throwable cause) {
    super(errorMsg, cause);
  }

  @Override
  public int restStatusCode() {
    return HttpServletResponse.SC_NOT_FOUND;
  }
}
```

**Review by chuckluv — DISMISSED** [review]

File: frontend/src/main/java/com/google/tripmeout/frontend/error/PlaceVisitAlreadyExistsException.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import com.google.tripmeout.frontend.error.TripMeOutException;
+import java.lang.Throwable;
+import javax.servlet.http.HttpServletResponse;
+
+
+/**
+ * exception to throw if a PlaceVisitModel object already exists in storage
+ */
+public class PlaceVisitAlreadyExistsException extends TripMeOutException {
+  public PlaceVisitAlreadyExistsException(String errorMssg) {
```
Comment (KoopaKing) [inline frontend/src/main/java/com/google/tripmeout/frontend/error/PlaceVisitAlreadyExistsException.java]: Why mssg tho?

**Review by KoopaKing — COMMENTED** [review]
LGTM other than mssg....

File: frontend/src/main/java/com/google/tripmeout/frontend/error/PlaceVisitAlreadyExistsException.java (hunk 1 above)
Comment (KoopaKing) [reply frontend/src/main/java/com/google/tripmeout/frontend/error/PlaceVisitAlreadyExistsException.java]: Actually this is really a nit, so whatever.

**Review by KoopaKing — DISMISSED** [review]

**Review by KoopaKing — APPROVED** [review]

**Review by anfee1 — APPROVED** [review]
LGTM

**Pull request merged**